tests/unit/mixins/should-add-token-test.js

```diff
@@ -47,3 +46,59 @@ test('does throw an exception with bad input', function (assert) {
   assert.notOk(subject.hostsMatch(undefined, 'https://server.foo.bar.com:6443/FeatureServer/0'));
   assert.notOk(subject.hostsMatch(undefined, {}));
 });
+
+test('shouldAddToken returns false when token should not be added', function (assert) {
+  const url = 'https://services2.arcgis.com/zNjnZafDYCAJAbN0/arcgis/rest/services/Street_ROW_Trees/FeatureServer/0/query?returnCountOnly=true&f=json&where=1%3D1&inSR=4326&geometry=-118.132,34.16,-118.122,34.162&geometryType=esriGeometryEnvelope&spatialRel=esriSpatialRelIntersects';
+  const serverInfo = {
+    owningSystemUrl: 'http://www.arcgis.com',
+    owningTenant: 'zNjnZafDYCAJAbN0'
+  };
+  const portalInfo = {
+    portalHostname: 'qaext.arcgis.com',
+    id: '97KLIFOSt5CxbiRI',
+    authorizedCrossOriginDomains: []
+  };
+  assert.notOk(subject.shouldAddToken(url, serverInfo, portalInfo));
+});
+
+test('shouldAddToken returns true when portal ids match', function (assert) {
+  const url = 'https://servicesqa.arcgis.com/97KLIFOSt5CxbiRI/ArcGIS/rest/services/integration_features_10001/FeatureServer/0/query?returnCountOnly=true&f=json&where=1%3D1&inSR=4326&geometry=-95.107,41.389,-95.025,41.4&geometryType=esriGeometryEnvelope&spatialRel=esriSpatialRelIntersects';
+  const serverInfo = {
+    owningSystemUrl: 'http://www.arcgis.com',
+    owningTenant: '97KLIFOSt5CxbiRI'
+  };
+  const portalInfo = {
+    portalHostname: 'qaext.arcgis.com',
+    id: '97KLIFOSt5CxbiRI',
+    authorizedCrossOriginDomains: []
+  };
+  assert.ok(subject.shouldAddToken(url, serverInfo, portalInfo));
```

Comment:
I'm surprised this works given that servicesqa is not included as an authorized url and servicesqa.arcgis.com !== www.arcgis.com

Reply:
In this case it is because `serverInfo.owningTenant === portalInfo.id`

Reply:
thats bad because the portals aren't the same. I don't think we should be relying on portal id at all.

Reply:
I can take that out but is seems correct (though probably unnecessary) to me.

Reply:
I don't think it's correct. portal: qaext id: 123 !== portal: devext  id: 123